service/data/edgedb/agreements.ts

```diff
@@ -9,359 +11,230 @@ import {
   AgreementVersion,
   AgreementTextInput,
 } from "../../domain/agreements";
-import {ServerError} from "../../common/app";
-
-// The following interfaces describe the shape of DB entities,
-
-interface TextEntity {
-  id: string;
-  title: string;
-  text: string;
-  culture: string;
-  update_time: Date;
-  creation_time: Date;
-}
-
-interface VersionEntity {
-  id: string;
-  current: boolean;
-  draft: boolean;
-  creation_time: string;
-  agreement_id?: string[];
-  texts?: TextEntity[];
-}
-
-function mapTextEntity(entity: TextEntity): AgreementText {
-  return new AgreementText(
-    entity.id,
-    entity.title,
-    entity.text,
-    entity.culture,
-    "",
-    new Date(entity.update_time),
-    new Date(entity.creation_time)
-  );
-}
-
-function mapVersionEntity(entity: VersionEntity): AgreementVersion {
-  return new AgreementVersion(
-    entity.id,
-    entity.current,
-    entity.draft,
-    entity.agreement_id ? entity.agreement_id[0] : undefined,
-    new Date(entity.creation_time),
-    entity.texts ? entity.texts.map(mapTextEntity) : undefined
-  );
-}
 
 @injectable()
 export class EdgeDBAgreementsRepository
   extends EdgeDBRepository
   implements AgreementsRepository
 {
   async getAgreement(agreementId: string): Promise<Agreement | null> {
-    const agreement = await this.run(async (connection) => {
-      return await connection.querySingle<{
-        id: string;
-        name: string;
-        description: string;
-        creation_time: Date;
-        versions: VersionEntity[];
-      }>(
-        `SELECT Agreement {
-          name,
-          description,
-          creation_time,
-          versions: {
-            current,
-            draft,
-            creation_time
-          } ORDER BY .current DESC
-        }  FILTER .id = <uuid>$id;`,
-        {
-          id: agreementId,
-        }
-      );
-    });
-
-    if (!agreement) return null;
-
-    return new Agreement(
-      agreement.id,
-      agreement.name,
-      agreement.description,
-      agreement.creation_time,
-      agreement.versions.map(mapVersionEntity)
+    return await this.run(async (connection) =>
+      e
+        .select(e.Agreement, (agreement) => ({
+          id: true,
+          name: true,
+          description: true,
+          creationTime: agreement.creation_time,
```

Comment:
This is a cool pattern for rewriting snake case to camel case. Going to file this away.